### PR TITLE
add HANA scale-out deployments for AWS/GCP/libvirt/openstack

### PR DESCRIPTION
**Pull request by yeoldegrove**
## prerequisites
* needs commits from #845 
* needs https://github.com/SUSE/shaptools/pull/70
* needs https://github.com/SUSE/salt-shaptools/pull/88

## features
* add HANA scale-out deployments for AWS/GCP/libvirt
  * AWS -> on shared EFS storage
    * https://docs.aws.amazon.com/sap/latest/sap-hana/operating-system-and-storage-configuration.html
    * https://documentation.suse.com/sbp/all/html/SLES-SAP-hana-scaleOut-PerfOpt-12-AWS/index.html
  * GCP -> on shared filestore storage
    * https://cloud.google.com/solutions/sap/docs/sap-hana-ha-scaleout-deployment-guide
    * https://cloud.google.com/filestore/docs/creating-instances
  * libvirt -> on shared NFS storage

### CODE REVIEW

File: aws/terraform.tfvars.example [hunk 1]
```diff
@@ -137,16 +137,63 @@ cluster_ssh_key = "salt://sshkeys/cluster.id_rsa"
...
+# Number of standby nodes per site (not recommended on AWS)
+#hana_scale_out_standby_count = 0 # default:0 - Deploy X standby nodes per site. The rest of the nodes will be worker nodes.
+
+#hana_majority_maker_instancetype = "t3.micro"
+#hana_majority_maker_ip =  "10.0.3.9"
+
+#########################
+# shared storage variables
+# Needed if HANA is deployed in scale-out scenario
+# see https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/hana-vm-operations-netapp
```
Comment (melzer-b1) [inline aws/terraform.tfvars.example:166]: this is an AWS example

Comment (yeoldegrove) [reply aws/terraform.tfvars.example:166]: fixed and force pushed

File: openstack/README.md [hunk 2]
```diff
@@ -124,4 +124,54 @@ the `terraform.tfvars` file and adjusting some variables. The example of how to
...
+  lv_sizes         = "100#100#100#100#100"
+  paths            = "/hana/data#/hana/log#/hana/shared#/usr/sap#/hana/backup"
+}
+```
+
+During deployment, HANA VM expects a standard set of directories for its data storage `/hana/data`, `/hana/log`, `/hana/shared`, `/usr/sap` and `/hana/backup`. 
+
+A HANA VM typically uses 5 to 10 disks according to usage scenario. These are combined to several logical volumes. At last the data locations of the standard mount points are assigned to these logical volumes.
+
+The first parameter `disks_size` is used to provision the resourcen in terraform. One disk is using one entry. Every further disk is added by appending more comma separated entries to each parameter.
```
Comment (melzer-b1) [inline openstack/README.md:149]: resources

File: aws/README.md [hunk 3]
```diff
@@ -233,7 +233,58 @@ Will deploy in Frankfurt 1 `t2.micro` instance as an iSCSI server, and 4 `m4.2xl
...
+  lv_sizes         = "100#100#100#100#100"
+  paths            = "/hana/data#/hana/log#/hana/shared#/usr/sap#/hana/backup"
+}
+```
+
+During deployment, HANA VM expects a standard set of directories for its data storage `/hana/data`, `/hana/log`, `/hana/shared`, `/usr/sap` and `/hana/backup`. 
+
+A HANA VM typically uses 5 to 10 disks according to usage scenario. These are combined to several logical volumes. At last the data locations of the standard mount points are assigned to these logical volumes.
+
+The first two parameters `disks_type` and `disks_size` are used to provision the resourcen in terraform. One disk is using one entry. Every further disk is added by appending more comma separated entries to each parameter.
```
Comment (melzer-b1) [inline aws/README.md:258]: resources

**Review by melzer-b1 — CHANGES_REQUESTED** [review]

File: openstack/README.md (hunk 2 above)
Comment (yeoldegrove) [reply openstack/README.md:149]: fixed

File: libvirt/modules/majority_maker_node/salt_provisioner.tf [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+resource "null_resource" "wait_after_cloud_init" {
+  count = var.common_variables["provisioner"] == "salt" ? var.node_count : 0
+
+  triggers = {
+    majority_maker_ids = libvirt_domain.majority_maker_domain[count.index].id
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "if which cloud-init; then cloud-init status --wait; else echo no cloud-init installed; fi"
```
[the remark this answers is not in the document]
Comment (yeoldegrove) [reply libvirt/modules/majority_maker_node/salt_provisioner.tf]: fixed

File: openstack/README.md (hunk 2 above)
Comment (yeoldegrove) [reply openstack/README.md:149]: will be fixed in #845

File: aws/README.md (hunk 3 above)
Comment (yeoldegrove) [reply aws/README.md:258]: will be fixed in #845

**Review by melzer-b1 — APPROVED** [review]

**Pull request merged**